velox/expression/SimpleFunctionRegistry.cpp

```diff
@@ -60,6 +60,9 @@ bool SimpleFunctionRegistry::registerFunctionInternal(
           metadata->defaultNullBehavior(), otherMetadata.defaultNullBehavior());
       VELOX_USER_CHECK_EQ(
           metadata->isDeterministic(), otherMetadata.isDeterministic());
+      VELOX_USER_CHECK_EQ(
+          metadata->signature()->variableArity(),
```

Comment:
This is a separate change, right? Perhaps, it can be extracted into a separate PR?

Reply:
Before this Patch, signature is the key used for comparison, `variableArity` is always equal in that case, after this change, we don't allow to register `func(int)` and `func(int...)` at the same time.

Reply:
Sounds good. Would you extract this change into a separate PR?

Reply:
Sure.

Reply:
Added https://github.com/facebookincubator/velox/pull/10980